File: back/src/main/java/com/back/services/TaskService.java
package com.back.services;
import java.util.List;
import java.util.Map;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;
import org.springframework.web.bind.annotation.RequestBody;
import com.back.entities.*;
import com.back.repositories.FolderRepository;
import com.back.repositories.TaskRepository;

@Service
public class TaskService {
	// Task service
	@Autowired
	private TaskRepository taskRepo;
	@Autowired
	private FolderRepository folderRepo;
	
	// Returns all the task related to the folder related to the given id
	public List<Task> getTasks(Long id) {
		return folderRepo.getById(id).getTasks();
	}
	
	// Returns the task related to the given id
	public Task getTaskById(Long id) {
		Task task = taskRepo.findById(Long.valueOf(id)).get();
		return task;
	}
	
	// Create a task with the data received in the payload
	public void createTask(@RequestBody Map<String, Object> payload) {
		Long folder_id = Long.valueOf((String)payload.get("folder_id"));
		String content = (String)payload.get("content");
		Folder folder = folderRepo.getById(folder_id);
		Task newTask = new Task();
		newTask.setContent(content);
		newTask.setDone(false);
		taskRepo.save(newTask);
		folder.getTasks().add(newTask);
		folderRepo.save(folder);
	}
	
	// Edit a task with the data received in the payload
	public void editTask(@RequestBody Map<String, Object> payload) {
		Long id = Long.valueOf((String)payload.get("id"));
		String content = (String)payload.get("content");
		Task task = taskRepo.getById(id);
		task.setContent(content);
		taskRepo.save(task);
	}
	
	// Switch the 'done' state of the task related to the given id
	public void switchStateTask(Long id) {
		Task task = taskRepo.getById(Long.valueOf(id));
		task.setDone(!task.isDone());
		taskRepo.save(task);
	}
	
	// Deletes the task related to the given id
	public void deleteTask(Long id) {
		Task task = taskRepo.getById(Long.valueOf(id));
		taskRepo.delete(task);
	}
}

File: front/src/services/TaskService.js
import axios from 'axios';
const ApiURL = 'http://localhost:8080/api/';

// Service for tasks
class TaskService {
    // Get the tasks related to the folder from the database
    getTasks(folder_id) {
        return axios.get(ApiURL + 'tasks/' + folder_id);
    }

    // Create a new task in the database
    createTask(task) {
        return axios.post(ApiURL + 'newTask', task);
    }

    // Get the task related to the given id from the database
    getTaskById(id) {
        return axios.get(ApiURL + 'task/' + id);
    }

    // Edit the given task
    editTask(task) {
        return axios.post(ApiURL + 'editTask', task);
    }

    // Switch the state 'done' of the given task
    completeTask(task) {
        return axios.post(ApiURL + 'switchStateTask', {
            id: task.id,
        });
    }

    // Delete the task with the given id from the database
    deleteTask(id) {
        return axios.delete(ApiURL + 'deleteTask/' + id);
    }
}

export default new TaskService();